### PR TITLE
Future regressors modularization

### PR DESCRIPTION
**Pull request by alfonsogarciadecorral**
## :microscope: Background

- Following the example of trend, the future regressors component is modularized.

## :crystal_ball: Key changes

- Created a new Seasonality abstract and FourierSeasonality class.

## :clipboard: Review Checklist
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, added docstrings and data types to function definitions.

Please make sure to follow our best practices in the [Contributing guidelines](https://github.com/ourownstory/neural_prophet/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: neuralprophet/plot_utils.py [hunk 1]
```diff
@@ -465,7 +465,7 @@ def get_valid_configuration(  # move to utils
                     }
                 )
             elif validator == "plot_parameters":
-                regressor_param = m.model.get_reg_weights(regressor)[quantile_index, :]
+                regressor_param = m.model.future_regressors.get_reg_weights(regressor)[quantile_index, :]
```
Comment (karl-richter) [inline neuralprophet/plot_utils.py:468]: sweet, like that you include the regularization already!

File: neuralprophet/time_net.py [hunk 2]
```diff
@@ -347,30 +329,18 @@ def __init__(
...
         # Regressors
+        # Migrating Regressors
         self.config_regressors = config_regressors
-        self.regressors_dims = utils.config_regressors_to_model_dims(config_regressors)
-        if self.regressors_dims is not None:
-            n_additive_regressor_params = 0
-            n_multiplicative_regressor_params = 0
-            for name, configs in self.regressors_dims.items():
-                if configs["mode"] not in ["additive", "multiplicative"]:
-                    log.error("Regressors mode {} not implemented. Defaulting to 'additive'.".format(configs["mode"]))
```
Comment (karl-richter) [inline neuralprophet/time_net.py:357]: Just a comment: We should make a consistent decision where to raise error messages if certain parameters are missing. But fine for now.

File: neuralprophet/time_net.py [hunk 3]
```diff
@@ -832,26 +773,24 @@ def compute_components(self, inputs, meta):
                 )
         if self.config_regressors is not None and "regressors" in inputs:
             if "additive" in inputs["regressors"].keys():
-                components["future_regressors_additive"] = self.scalar_features_effects(
-                    features=inputs["regressors"]["additive"], params=self.regressor_params["additive"]
```
Comment (karl-richter) [inline neuralprophet/time_net.py:836]: Really good solution, I like how much simpler the code now is!

File: neuralprophet/time_net.py [hunk 4]
```diff
@@ -832,26 +773,24 @@ def compute_components(self, inputs, meta):
...
                 if mode == "additive":
                     features = inputs["regressors"]["additive"]
-                    params = self.regressor_params["additive"]
+                    mode = "additive"
                 else:
                     features = inputs["regressors"]["multiplicative"]
-                    params = self.regressor_params["multiplicative"]
-                components[f"future_regressor_{regressor}"] = self.scalar_features_effects(
-                    features=features, params=params, indices=index
-                )
```
Comment (karl-richter) [inline neuralprophet/time_net.py:854]: What exactly is this section of the new code doing? The following looks kind of redundant
```python
if mode == "additive":
    ...
    mode = "additive"
```

File: neuralprophet/utils_torch.py [hunk 5]
```diff
@@ -10,6 +10,25 @@
...
+    """
+    if len(dims) > 1:
+        return nn.Parameter(nn.init.xavier_normal_(torch.randn(dims)), requires_grad=True)
+    else:
+        return nn.Parameter(torch.nn.init.xavier_normal_(torch.randn([1] + dims)).squeeze(0), requires_grad=True)
+
+
 def init_parameter(dims):
     """
     Create and initialize a new torch Parameter.
```
Comment (karl-richter) [inline neuralprophet/utils_torch.py:34]: I think we did the same thing here ;)

**Review by karl-richter — APPROVED** [review]
There's one if-else statement where I suspect some dublicate, please review and change if applicable. Otherwise looks really good, ready to go from my side.

File: neuralprophet/utils_torch.py (hunk 5 above)
Comment (karl-richter) [reply neuralprophet/utils_torch.py]: Changed this in the latest commit

File: neuralprophet/time_net.py (hunk 4 above)
Comment (karl-richter) [reply neuralprophet/time_net.py:854]: @alfonsogarciadecorral can you have a quick look at this?

**Review by ourownstory — APPROVED** [review]
Great Work!!

File: neuralprophet/components/future_regressors/future_regressors.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+from neuralprophet import utils
+from neuralprophet.components import BaseComponent
+
+
+class FutureRegressors(BaseComponent):
+    def __init__(self, config, id_list, quantiles, n_forecasts, device, log, config_trend_none_bool):
```
Comment (ourownstory) [inline neuralprophet/components/future_regressors/future_regressors.py:6]: the logger can be retrived at the beginning of a file - does not need to be passed by. (see examples of other files)

**Review by ourownstory — CHANGES_REQUESTED** [review]
only issue: let's not pass logger

**Review by ourownstory — APPROVED** [review]

**Pull request merged**